app/models/pipeline_run.rb

```diff
@@ -65,15 +66,18 @@ def local_json_path
   end
 
   def load_results_from_s3
-    return if pipeline_output
+    return if pipeline_output && !pipeline_output.taxon_byteranges.empty?
```

Comment:
Do you have an extra `!` here?   If the byte ranges have not been computed, don't you want to compute them?

Reply:
I think the logic is correct. If taxon_byteranges already exist ("not empty"), the function returns early. If taxon_byteranges are empty, the function loads them in. BUT typing up this answer I realize I don't need to check taxon_byteranges at all; checking pipeline_output is enough because a new pipeline_run (even for a sample that has been run before) will always have no pipeline_output at first.